hydra-cluster/exe/hydra-cluster/Main.hs

```diff
@@ -24,13 +26,21 @@ run options =
     withStateDirectory $ \workDir ->
       case knownNetwork of
         Just network -> do
-          when (useMithril == UseMithril) $
-            downloadLatestSnapshotTo (contramap FromMithril tracer) network workDir
-          withCardanoNodeOnKnownNetwork fromCardanoNode workDir network $ \node -> do
-            waitForFullySynchronized fromCardanoNode node
-            publishOrReuseHydraScripts tracer node
-              >>= singlePartyHeadFullLifeCycle tracer workDir node
-        Nothing ->
+          maybeRunningNode <- findRunningCardanoNode workDir
+          case maybeRunningNode of
+            Nothing -> do
+              when (useMithril == UseMithril) $
+                downloadLatestSnapshotTo (contramap FromMithril tracer) network workDir
```

Comment:
Must: delete the `db/` directory here if you move it out of the workflow

In principle a good idea to not do it in the workflow, but it's nowhere deleted now so this will never work (mithril fails if the `db` dir already exists)